.circleci/config.yml

```diff
@@ -0,0 +1,101 @@
+version: 2.1
+commands:
+  restore-cache:
+    steps:
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package-lock.json" }}
+            - v1-dependencies-
+  install-dependencies:
+    steps: 
+      - run: npm install
+  save-cache:
+    steps:
+      - save_cache:
+          key: v1-dependencies-{{ checksum "package-lock.json" }}
+          paths:
+            - ./node_modules
+  generate-prisma-client:
+    steps:
+      - run: npx prisma generate
+  generate-prisma-migrations:
+    steps:
+      - run: npx prisma migrate dev
+  deploy-prisma-migrations:
+    steps:
+      - run: npx prisma migrate deploy --preview-feature
+  run-tests:
+    steps:
+      - run: npm run test:ci 
+      - run: npm run test:e2e 
+  greetings-from-branch:
+    steps:
+      - run: echo "No siema z brancha <<pipeline.git.branch>>"
+executors:
+  executor:
```

Comment:
when you define executor like this and reuse it in other places keep in mind that you will run node and postgress everywhere.
You do not need postgress in ` greetings-from-branch:` 
you could either define multiple executor combinations or by adding some reusable code (aliastes) with that will be inserted. Like so:

![image](https://github.com/marybrown123/recipes-api/assets/52794342/3e7bca1c-c161-46e6-bcbf-9ec7d23b5fd1)

![image](https://github.com/marybrown123/recipes-api/assets/52794342/11780076-bacc-4068-b58f-307c46f6a24f)


https://circleci.com/docs/introduction-to-yaml-configurations/#merging-maps